app/models/validations/sales/property_validations.rb

```diff
@@ -7,4 +7,13 @@ def validate_postcodes_match_if_discounted_ownership(record)
       record.errors.add :ppostcode_full, I18n.t("validations.property.postcode.must_match_previous")
     end
   end
+
+  def validate_bedsit_number_of_beds(record)
+    return if record.is_bedsit?.blank? || record.beds.blank?
```

Comment:
aha. I see related to my commetn above, that it would return nil is proptype is nil and then this would work as a guard clause.
I think it would work better in being consistent with the other validations for this line to be ` return if record.proptype.blank? || ...`

separately this could be `return unless record.proptype && record.beds` which is slightly more concise but they would both be doing the same thing so it's really a subjective readability choice and I don't see a clear convention on the project.

Reply:
Thanks @arfacamble. It seems odd to me to go to the trouble of writing an `is_bedsit?` method and then use it only on line 14 here and not in the logical evaluation on line 12 - what do you think?

Reply:
I think it's good to have it for line 14 as it makes that line more readable, even if it's only used in that one place.

I think it's a bit misleading on line 12 because the convention for a method ending in ? is that it returns true or false.
Due to the thruthy falsey nature of ruby, I think it's just about fine for that method to return nil when it means false, but I don't think that should be intended to be used as a 'nil' because then it goes against that contravention.

On that line 12 I'm kind of expecting to be calling `false.blank?` or `true.blank?` which again will work but is on the limits of having a bit of a code smell. I have to think harder to work out what will happen in each situation. In the end it boils down to returning if proptype is not bedsit, si I think it would be better if you want to use it, to have !record.is_bedsit?

Reply:
Could it be `if record.is_bedsit? &&  record.beds &&  record.beds > 1` on line 14 and we remove the guard clause?

Reply:
more concise formulation:

in general for these validations and this follows a pattern more generally, these guard clauses aren't really for the proper logic, they're guarding against errors (and excluding cases we don't care about by so doing).

generally these guard clauses across these methods are ways of avoiding get nil reference exceptions by eg trying to compare nil to 1 below

Reply:
Thanks both :) I like the idea of keeping the guard clause, but of course making it a bit more readable and in line with convention. Have done so - lmk if you have any other suggestions. Thanks!